app/models/manageiq/providers/ansible_tower/shared/automation_manager/tower_api.rb

```diff
@@ -25,6 +25,26 @@ def create_in_provider_queue(manager_id, params, auth_user = nil)
       queue(manager.my_zone, nil, "create_in_provider", [manager_id, params], action, auth_user)
     end
 
+    def launch_in_provider(result_id)
+      template = ConfigurationScript.find(result_id)
+      template.with_provider_object do |object|
+        job = object.launch
+        job.to_hash.to_json
+      end
+    rescue AnsibleTowerClient::ClientError => error
```

Comment:
I think you'll need to rescue  `ActiveRecord::RecordNotFound` in case the `find` fails.